test/components/pagination.test.tsx

```diff
@@ -1,28 +1,39 @@
 import React from 'react';
 import renderer from 'react-test-renderer';
 import Pagination from '../../src/components/pagination';
+import { createPaginationInfo } from '../__fixtures__/page';
 
 describe('Pagination component', () => {
   it('renders links to the next and previous page', () => {
-    const next = { frontmatter: { title: 'next-page' }, fields: { slug: 'next-page' }};
-    const prev = { frontmatter: { title: 'prev-page' }, fields: { slug: 'prev-page' }};
-    const tree = renderer.create(<Pagination next={next} previous={prev} />).toJSON();
+    const paginationInfo = createPaginationInfo();
```

Comment:
nit: Since `const paginationInfo = createPaginationInfo();` is a setup action that needs to be done before each test, we could use jest's [`beforeEach`](https://jestjs.io/docs/en/setup-teardown) method to do so.

And so, our describe block would look like

```ts
describe('Pagination component', () => {
  let paginationInfo;

  beforeEach(() => {
    paginationInfo = createPaginationInfo();
  });

  it('renders links to the next and previous page', () => {
    const tree = renderer
      .create(<Pagination next={paginationInfo} previous={paginationInfo} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when there is no next page', () => {
    const tree = renderer
      .create(<Pagination previous={paginationInfo} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when there is no previous page', () => {
    const tree = renderer.create(<Pagination next={paginationInfo} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('only renders links to pages that has a title', () => {
    const nextPaginationInfo = { slug: 'test-slug', title: null };
    const tree = renderer
      .create(
        <Pagination next={nextPaginationInfo} previous={paginationInfo} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
```

Reply:
I get your point with this, but in my opinion, it's better to be explicit and not share any state between tests when possible. I mean we have already abstracted away the actual setup action. Have a look [here](https://github.com/jest-community/eslint-plugin-jest/blob/2b55a863cd092537abc9c89992ccf379671a4322/docs/rules/no-hooks.md) for some input from the jest community on this 🙂

Reply:
That's great input, Olle. I see your point ([jamiebuilds](https://github.com/jamiebuilds) makes some [great points](https://github.com/avajs/ava/issues/222#issuecomment-365698996) in support of this).

Let's keep these tests as they are.